lib/rmt/cli/mirror.rb

```diff
@@ -34,60 +31,66 @@ def all
   desc 'repository IDS', _('Mirror enabled repositories with given repository IDs')
   def repository(*ids)
     RMT::Lockfile.lock('mirror') do
-      logger = RMT::Logger.new(STDOUT)
-      mirror = RMT::Mirror.new(logger: logger, mirror_src: RMT::Config.mirror_src_files?)
-
       ids = clean_target_input(ids)
       raise RMT::CLI::Error.new(_('No repository IDs supplied')) if ids.empty?
 
-      repos = []
-      ids.each do |id|
-        repo = Repository.find_by!(scc_id: id)
-        raise RMT::CLI::Error.new(_('Mirroring of repository with ID %{repo_id} is not enabled') % { repo_id: id }) unless repo.mirroring_enabled
-        repos << repo
-      rescue ActiveRecord::RecordNotFound
-        raise RMT::CLI::Error.new(_('Repository with ID %{repo_id} not found') % { repo_id: id })
+      repos = ids.map do |id|
+        repo = Repository.find_by(scc_id: id)
+        errors << _('Repository with ID %{repo_id} not found') % { repo_id: id } if repo.nil?
+
+        repo
       end
 
-      _, repo_errors = mirror_repos!(mirror, repos)
-      handle_errors!(repo_errors)
+      mirror_repos!(mirror, repos)
```

Comment:
I'm not sure that `mirror` needs to be passed anymore.

Reply:
I really forgot this one. Fixed :heavy_check_mark: